src/main/java/seedu/linkedout/model/applicant/SortComparator.java

```diff
@@ -0,0 +1,76 @@
+package seedu.linkedout.model.applicant;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+
+public class SortComparator implements Comparator<Applicant> {
+
+    private final Field field;
+    private final Order order;
+
+    /**
+     *
+     * @param field
+     * @param order
+     */
+    public SortComparator(Field field, Order order) {
+        // assume both pass parseUtil before passing here
+        requireNonNull(field);
+        requireNonNull(order);
+        this.field = field;
+        this.order = order;
+    }
+
+    @Override
+    public int compare(Applicant applicant1, Applicant applicant2) {
+        requireNonNull(applicant1);
+        requireNonNull(applicant2);
+        String orderString = this.order.toString();
+        String fieldString = this.field.toString();
+        switch(orderString) {
+        case "ASC":
+            switch (fieldString) {
+            case "NAME":
+                return applicant1.compareNames(applicant2);
```

Comment:
In line with the other comment,

Line 35 should be something like `applicant1.getName().compareTo(applicant2.getName())`

Reply:
I think I cant really override it inside Applicant, as that is for comparing flag status, hence it will be a Name implements Comparable<Name> instead now @RyanCheungJF